Pi_Code/Controller_Operations.py

```diff
@@ -0,0 +1,113 @@
+from multiprocessing import Process
+import pygame
+
+# Axes legend:
+# Axes 0: Roll (right is positive, left is negative)
+# Axes 1: Pitch (back is positive, forward is negative)
+# Axes 2: Throttle (0% position is positive, 100% position is negative)
+# Axes 3: Yaw (clockwise is positive, counterclockwise is negative)
+# Button legend:
+# Button 0: Trigger
+# Button 6: Labelled as 07 on the joystick
+# Button 5: Labelled as 06 on the joystick
+# Button 1: Labelled as 02 on the joystick
+
+
+class ControllerOps:
+
+    def __init__(self, dataFile, connection, SaveSensorData):
+        # We need to display a window in order to read joystick inputs.
+        pygame.joystick.init()
+        screen = pygame.display.set_mode((10, 10))
+
+        self.controller = pygame.joystick.Joystick(0)
+        self.controller.init()
+
+        self.idle = False # TODO: Change idle to true and add state switching in the main Submarine script
+        # Currently we'll just skip right past the idle stage and right into the auto sub stage. Idle is intended to be
+		# where we click the "start" button on the joystick to begin the operation of the sub
+        self.read_stick = True
+        self.autosub = True
+
+        # Get the properties of the joystick
+        # Not necessary currently, may be useful later
+        print("Found joystick: " + self.controller.get_name())
+        buttons = self.controller.get_numbuttons()
+        axes = self.controller.get_numaxes()
+        balls = self.controller.get_numballs()
+        hats = self.controller.get_numhats()
+        p = Process(target = self.getStickPosition, args = (dataFile, connection, SaveSensorData))
+        p.start()
+
+    def getStickPosition(self, connection, dataFile, SaveSensorData):
+
+        while self.idle:
```

Comment:
I don't get this switching, you are stuck in idle until you go to auto in which you are stuck until you go back into the manual mode and then you will start getting values?

Did you mean ot have this be an in statement?

Reply:
Intent is that we use the joystick as the start button for the sub. It starts with the sub doing nothing, we click the button and it sets a flag in Submarine.py that beings the autonomous system. If we want to instead control it manually then we can press button 5, which tells the autonomous system to let us manually control it. 

I didn't do more than this in implementing it because you know the code better than I do, and I wanted to go over the process with you before I did anything more than this.